install-and-configure/install/multi-cluster/federated-etl/aggregator.md

```diff
@@ -69,88 +69,69 @@ Because the Aggregator PV is relatively small, the least expensive performance g
-
-  # this is the number of partitions the datastore is split into for copying
-  # the higher this number, the lower the ram usage but the longer it takes for
-  # new data to show in the kubecost UI
-  # set to 0 for max partitioning (minimum possible ram usage, but the slowest)
-  # the default of 25 is sufficient for 95%+ of users. This should only be modified
-  # after consulting with Kubecost's support team
-  numDBCopyPartitions: 1
   logLevel: info
 
-  # env: has been removed to avoid unknown issues that would be caused by
-  # customizations that were required to run aggregator in previous versions
-  # extraEnv: can be used to add new environment variables to the aggregator pod
-
-  # the below settings should only be modified with support from Kubecost staff
+  # How much data to ingest from the federated store bucket, and how much data
+  # to keep in the DB before rolling the data off.
+  # 
+  # Note: If increasing this value to backfill historical data, it will take
+  # time to gradually ingest & process those historical ETL files. Consider
+  # also increasing the resources available to the aggregator as well as the
+  # dbConcurrentIngestionCount.
+  # 
+  # default: 91
+  etlDailyStoreDurationDays: 365
 
   # How many threads the read database is configured with (i.e. Kubecost API /
   # UI queries). If increasing this value, it is recommended to increase the
   # aggregator's memory requests & limits.
   # default: 1
   dbReadThreads: 1
+
   # How many threads the write database is configured with (i.e. ingestion of
   # new data from S3). If increasing this value, it is recommended to increase
   # the aggregator's memory requests & limits.
   # default: 1
   dbWriteThreads: 1
+
   # How many threads to use when ingesting Asset/Allocation/CloudCost data
   # from the federated store bucket. In most cases the default is sufficient,
   # but can be increased if trying to backfill historical data.
   # default: 1
   dbConcurrentIngestionCount: 1
-  # dbCopyFull: "true" can improve the time it takes to copy the write DB, 
-  # at the expense of additional memory usages.
-  dbCopyFull: "false"
+
   # Memory limit applied to read database connections.
   # default: 0Gi is no limit
   dbMemoryLimit: 0Gi
+
   # Memory limit applied to write database connections.
   # default: 0Gi is no limit
   dbWriteMemoryLimit: 0Gi
-  # How much data to ingest from the federated store bucket, and how much data
-  # to keep in the DB before rolling the data off.
-  # 
-  # Note: If increasing this value to backfill historical data, it will take
-  # time to gradually ingest & process those historical ETL files. Consider
-  # also increasing the resources available to the aggregator as well as the
-  # refresh & concurrency env vars.
-  # 
-  # default: 91
-  etlDayStoreDurationDays: 91
+
+  # If "true" can improve the time it takes to copy the write DB, at the expense
+  # of additional memory usage.
+  # default: "false"
+  dbCopyFull: "false"
```

Comment:
@mbolt35 Based on my reading of the code, setting this to `true` would create a separate DB Transaction when copying each chunk/partition of the table. This is probably a safer and less memory intensive approach. But probably also slower?

Reply:
So, this was added as sort of a quick test, but I actually think it might be valuable going forward to have this option. I think this description is probably "ok" for now, although I'm not sure how _true_ it actually is. 

Both methods are _very_ similar, and my guess is that they likely have a fairly equivalent memory footprint, but it's actually possible that the non-full-copy option (`false`) will use more memory due to potential parallelization that can happen via db connections (likely depends on the db write threads > 1). 

To also provide some clear context on _why_ this option was created (hint: it wasn't added for memory/speed tradeoffs). We were running into some contention issues _during_ the copy, so this change was actually attempting to create a more atomic copy operation such that the contending reads on the old db didn't explode the copy (or vice-versa). 

For reference, here's the "full copy": 
https://github.com/kubecost/kubecost-cost-model/blob/305e43fd85a27867a7ca677c7a636283a2498204/pkg/duckdb/write/writer.go#L720-L953

It's a quick copy/paste job that just appends all the commands into a single SQL string and is issued over 1 db connection inside a txn. 

The default is actually a bit more "up in the air" in terms of the execution semantics (variable based on inputs):
https://github.com/kubecost/kubecost-cost-model/blob/305e43fd85a27867a7ca677c7a636283a2498204/pkg/duckdb/write/writer.go#L444-L718

The "chunking" in this case is sort of hand-wavy, as the window copies run serially, and the memory pool for the database will only cleanup when the full copy is complete. 

In conclusion here, I think we still want to look into:
* Is this option necessary at all, or can we find a very optimal copy mechanism for most/all configurations
* Are there even major differences between the two, and if so, what are they? (to be documented here). 

The contention issues have mostly been resolved (by completely avoiding reads on the write db during copy).

Reply:
Ok that context is really helpful. TLDR ... "it depends". Let's remove `dbCopyFull` and `numDBCopyPartitions` from docs and Helm defaults for now? We'll tune these as needed from a support perspective? @jessegoodier thoughts?

Reply:
I'm fine leaving in variables that we may want to be able to easily have customers change in the helm chart.
The comment already says:
# the below settings should only be modified with support from Kubecost staff

Reply:
@jessegoodier Ok I'm fine with approving and merging this PR to ensure docs are in alignment with the v2.3.1 release. However we need to return to this section of the docs and modify/update it as we learn more about how these tuning variables holistically affect Aggregator performance.